js/apps/system/_admin/aardvark/APP/frontend/js/views/graphManagementView.js

```diff
@@ -513,46 +602,73 @@
     },
 
     setFromAndTo: function (e) {
+      // This method triggers as soon as an edgeDefinition name is either being added or removed.
       e.stopPropagation();
-      var map = this.calculateEdgeDefinitionMap();
       var id;
 
-      const smartGraphActive = $('#tab-smartGraph').parent().hasClass('active');
-      const enterpriseGraphActive = $('#tab-enterpriseGraph').parent().hasClass('active');
-
-      if (smartGraphActive || enterpriseGraphActive) {
-        if (e.added) {
-          if (this.eCollList.indexOf(e.added.id) === -1 && this.removedECollList.indexOf(e.added.id) !== -1) {
-            id = e.currentTarget.id.split('row_newEdgeDefinitions')[1];
-            $('input[id*="newEdgeDefinitions' + id + '"]').select2('val', null);
-            $('input[id*="newEdgeDefinitions' + id + '"]').attr(
-              'placeholder', 'The collection ' + e.added.id + ' is already used.'
-            );
-            return;
-          }
-          this.removedECollList.push(e.added.id);
-          this.eCollList.splice(this.eCollList.indexOf(e.added.id), 1);
-        } else {
-          if (e.removed) {
-            // TODO edges not properly removed within selection
-            this.eCollList.push(e.removed.id);
-            this.removedECollList.splice(this.removedECollList.indexOf(e.removed.id), 1);
+      const getAllCurrentEdgeDefinitionNames = () => {
+        let edgeDefNames = [];
+        for (let i = 0; i <= this.counter; i++) {
+          let name = _.pluck($('#s2id_newEdgeDefinitions' + i).select2('data'), 'text')[0];
+          if (name) {
+            // might be undefined as we do not have the state of all real "counter" values
+            edgeDefNames.push(name);
           }
         }
-        if (map[e.val]) {
-          id = e.currentTarget.id.split('row_newEdgeDefinitions')[1];
-          $('#s2id_fromCollections' + id).select2('val', map[e.val].from);
-          $('#fromCollections' + id).attr('disabled', true);
-          $('#s2id_toCollections' + id).select2('val', map[e.val].to);
-          $('#toCollections' + id).attr('disabled', true);
-        } else {
+        return edgeDefNames;
+      };
+
+      const toFindDuplicates = (arr) => {
+        return arr.filter((item, index) => arr.indexOf(item) !== index);
```

Comment:
This is O(n^2). We could try to speed up the duplicate detection process.
```
const itemCounts = arr.reduce((acc, item) => {
  acc[item] = (acc[item] || 0) + 1;
  return acc;
}, {});

// return Object.keys(itemCounts).filter(key => itemCounts[key] > 1); // This could mess with the ordering of elements.
return _.chain(arr).filter(item => itemCounts[item] > 1).uniq().value(); // This preserves the original order, but is slightly more expensive due to the additional uniq() operation.
```

Reply:
Not that any of this matters for small arrays :laughing: